src/index/diskann/diskann.cc

```diff
@@ -32,7 +32,8 @@
 namespace knowhere {
 template <typename DataType>
 class DiskANNIndexNode : public IndexNode {
-    static_assert(std::is_same_v<DataType, fp32>, "DiskANN only support float");
+    static_assert(KnowhereFloatTypeCheck<DataType>::value,
+                  "DiskANN only support floating point data type(float32, float16, brain_float16)");
```

Comment:
Please use `bfloat16` instead of `brain_float16` :)

Reply:
updated